tensorboard/backend/event_processing/data_provider_test.py

```diff
@@ -103,45 +113,48 @@ def test_list_scalars_filters(self):
 
     result = provider.list_scalars(
         experiment_id="unused",
-        owner_plugin=scalar_metadata.PLUGIN_NAME,
+        plugin_name=scalar_metadata.PLUGIN_NAME,
         run_tag_filter=base_provider.RunTagFilter(runs=["waves", "hugs"]),
     )
     self.assertItemsEqual(result.keys(), ["waves"])
     self.assertItemsEqual(result["waves"].keys(), ["sine", "square"])
 
     result = provider.list_scalars(
         experiment_id="unused",
-        owner_plugin=scalar_metadata.PLUGIN_NAME,
+        plugin_name=scalar_metadata.PLUGIN_NAME,
         run_tag_filter=base_provider.RunTagFilter(["un"], ["likely"]),
     )
     self.assertEqual(result, {})
 
   def test_read_scalars(self):
-    provider = self.create_provider()
+    multiplexer = self.create_multiplexer()
+    provider = data_provider.MultiplexerDataProvider(multiplexer)
 
     run_tag_filter = base_provider.RunTagFilter(
-        runs=["waves", "polynomials"], tags=["sine", "square", "cube"]
+        runs=["waves", "polynomials", "unicorns"],
+        tags=["sine", "square", "cube", "iridescence"],
     )
-    step_filter = base_provider.StepFilter(lower_bound=2, upper_bound=-3)
-    # steps [0, 2, ..., 18] from 2 to -3 = [2, ..., 16]
     result = provider.read_scalars(
         experiment_id="unused",
-        owner_plugin=scalar_metadata.PLUGIN_NAME,
+        plugin_name=scalar_metadata.PLUGIN_NAME,
         run_tag_filter=run_tag_filter,
-        step_filter=step_filter,
-        downsample_to=None,  # not yet implemented
+        downsample=None,  # not yet implemented
     )
 
     self.assertItemsEqual(result.keys(), ["polynomials", "waves"])
     self.assertItemsEqual(result["polynomials"].keys(), ["square", "cube"])
     self.assertItemsEqual(result["waves"].keys(), ["square", "sine"])
-    sample_series = result["polynomials"]["square"]
-    self.assertLen(sample_series, 8)
-    sample = sample_series[6]
-    self.assertIsInstance(sample, base_provider.ScalarDatum)
-    self.assertEqual(sample.step, 14)
-    # nothing to test for wall time, as it can't be mocked out
-    self.assertEqual(sample.value, 49.0)
+    for run in result:
+      for tag in result[run]:
+        tensor_events = multiplexer.Tensors(run, tag)
+        self.assertLen(result[run][tag], len(tensor_events))
+        for (datum, event) in zip(result[run][tag], tensor_events):
+          self.assertEqual(datum.step, event.step)
+          self.assertEqual(datum.wall_time, event.wall_time)
+          self.assertEqual(
+              datum.value,
+              struct.unpack("<f", event.tensor_proto.tensor_content)[0],
```

Comment:
Optional, but IMO overall simpler if we always parse the tensors using `tensor_util.make_ndarray()` (or at least `np.frombuffer().item()`) even though it adds a dependency.

Reply:
Yes, I really should. :-) Done.